crypto/dso/dso_dlfcn.c

```diff
@@ -27,7 +27,33 @@
 #  endif
 #  include <dlfcn.h>
 #  define HAVE_DLINFO 1
-#  if defined(__SCO_VERSION__) || defined(_SCO_ELF) || \
+
+/* Cygwin initially did not support `dladdr`
+ * (as acknowledged by OpenSSL commit 73133962 on 2008-01-04
+ * and even earlier by OpenSSL commit 0ef888cd on 2006-04-11).
+ *
+ * Cygwin support for `dladdr` seems to have been added on 2017-03-06:
+ *   In the <https://cygwin.com/git/?p=newlib-cygwin.git;a=commit;h=c8432a01c8401c121940c806a9d868c4adc4cefd> gitweb page,
+ *   we can click on the *diff* links corresponding to:
+ *     - `winsup/cygwin/include/dlfcn.h`;
+ *     - `winsup/cygwin/include/cygwin/version.h`.
+ *
+ * OpenSSL commit 38f6f99c (pull request #9402 on 2019-07-17)
+ * takes advantage of newer, `dladdr`-supporting versions of Cygwin,
+ * but drops support for older, non-`dladdr`-supporting versions of Cygwin
+ * (e.g. as reported in issue #10200 on 2019-10-16).
+ *
+ * Unfortunately, Cygwin cannot generally be updated, as it periodically drops support
+ * for various versions of Windows (first XP, now all 32-bit and Vista, soon 7):
+ * <http://www.crouchingtigerhiddenfruitbat.org/cygwin/timemachine.html>.
+ *
+ * We now support both new and old versions of Cygwin
+ * by detecting `dladdr` support: if it exists, we continue to use it,
+ * but if it does not exist, we fallback to our old behaviour.
+ */
```

Comment:
I am not sure this lengthy commentary can be really regarded as something to be committed with CLA: trivial. Also IMO it does not belong here - I'd put it into the commit message and here just add one or two sentences.

Reply:
I believe that the evil practices of Cygwin -- forcing new versions of Windows upon us -- should be exposed visibly, in a comment in the source code. That is also going to prevent someone else from breaking support for older versions in the future.

Regarding the CLA: It is not like I am going to sue any organization because it has used my (possibly un-welcome anyway) lengthy comments (or that such a suing action would have success). This change is trivial -- one line of code (within a pre-processor directive, too).

The comments are meant to clarify the history of the situation. They can easily be ignored by Humans as they are ignored by the Compiler.

But the comments might help a Human who is interested in those lines of code -- to quickly get a picture of the situation without checking the Git log or the comments on the Github Pull Request webpage -- especially since Github itself might disappear -- just like support for old versions of Windows and Cygwin is disappearing -- Time beats us all. :'(

Is it possible to remove the "Requires CLA" mark, please ? (-:

Reply:
> Is it possible to remove the "Requires CLA" mark, please ? (-:

Unfortunately no. The requirements for CLA are strict, we do not allow much more than one or two line changes or similar trivial things without CLA.